### PR TITLE
feat(filterchecks): add optional metrics struct for caching

### PR DESCRIPTION
**Pull request by mstemm**
Add an optional metrics struct that if present, keeps track of the
efficacy of the extraction cache/evaluation cache.

The metrics count the number of times extract_cached() was called and
when it could use the cache, and the number of times comapre() was
called and when it could use the cache.

Like the caches, the memory for the pointer is assumed to be owned by
something else.

Signed-off-by: Mark Stemm <mark.stemm@gmail.com>

**What type of PR is this?**

> Uncomment one (or more) `/kind <>` lines:

> /kind bug

> /kind cleanup

> /kind design

> /kind documentation

> /kind failing-test

/kind feature

**Any specific area of the project related to this PR?**

> Uncomment one (or more) `/area <>` lines:

> /area build

> /area driver-kmod

> /area driver-ebpf

> /area libscap

/area libsinsp

> /area tests

> /area proposals

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
Add ability to track metrics on filtercheck cache efficacy. 
```

### CODE REVIEW

**Comment by FedeDP** [comment]
Hi Mark! Thank you, it LGTM!
We are in the process of freezing libraries for the next Falco 0.32.1 release; this can be merged right after :)

**Review by FedeDP — APPROVED** [review]
/approve

File: userspace/libsinsp/filterchecks.h
Comment (VadimZy) [inline userspace/libsinsp/filterchecks.h:77]: why class with all public why not struct then?

File: userspace/libsinsp/filterchecks.h
Comment (VadimZy) [inline userspace/libsinsp/filterchecks.h:184]: why is this naked ptr? who is managing its lifetime?

Comment (VadimZy) [reply userspace/libsinsp/filterchecks.h:184]: no NULLs in c++

**Review by FedeDP — APPROVED** [review]
/approve

**Comment by mstemm** [comment]
I'm going to remove the hold tag, as falco 0.32.1 has already been released. Still needs a second approval though.

**Comment by FedeDP** [comment]
Ups sorry! Go ahead!

**Review by Andreagit97 — APPROVED** [review]
/approve

**Pull request merged**
